chain/stmgr/call.go

```diff
@@ -140,25 +149,33 @@ func (sm *StateManager) CallWithGas(ctx context.Context, msg *types.Message, pri
 		// run the fork logic in `sm.TipSetState`. We need the _current_
```

Comment:
Is there a good reason why `Call` happens against tipset parent state and `CallWithGas` happens against tipset state?  As far as I understand is what requires the introduction of `hasExpensiveForkBetween`.  Beyond removing extra work it would also just be nice to standardize these two similar calls

Reply:
CallWithGas applies the message on-top-of the tipset to provide better gas estimation, I think. By executing it on-top-of the last tipset, instead of on-top-of the parent tipset state, we get the best gas estimates.

I'd prefer to always use the parent state, but I'm not sure what might break. I'm especially concerned that, e.g., it might (rarely) cause issues with window post:

1. See we're at height X.
2. Try to push the proof message.
3. Fail because the window isn't open yet.